contrib/mesos/pkg/minion/tasks/task.go

```diff
@@ -240,7 +247,8 @@ type exitError interface {
 	Sys() interface{}
 }
 
-func taskRunning(t *Task) taskStateFn {
+func taskRunning(t *Task) (nextState taskStateFn) {
```

Comment:
We have this nice state machine. Wouldn't it be clearer to have a state taskRunning and state taskGracefulTermination? Without this complicated for-loop magic?